File: number_theory/transform_prime.cpp
//{ Driver Code Starts
#include <bits/stdc++.h>
using namespace std;
#define ll long long
long long minNumber(long long arr[], long long N);

int main() {
  ios_base::sync_with_stdio(false);
  cin.tie(NULL);
  ll t;
  cin >> t;
  while (t--) {
    ll n;
    cin >> n;
    ll arr[n];
    for (ll i = 0; i < n; i++)
      cin >> arr[i];
    cout << minNumber(arr, n) << endl;
  }
  return 0;
}
// } Driver Code Ends

bool prime(long long n) {
  if (n < 2)
    return false;
  for (long long i = 2; i * i <= n; ++i) {
    if (n % i == 0)
      return false;
  }
  return true;
}

long long minNumber(long long arr[], long long N) {
  long long n = accumulate(arr, arr + N, 0LL);
  for (int i = n;; ++i) {
    if (prime(i))
      return i - n;
  }
  return -1;
}

File: number_theory/transform_prime.py
# User function Template for python3


class Solution:
    def minNumber(self, arr, n):
        # code here
        s = sum(arr)

        def prime(a):
            i = 2
            while i * i <= a:
                if a % i == 0:
                    return False
                i += 1

            return True

        i = s
        while True:
            if prime(i):
                return i - s
            i += 1

        return -1


# {
# Driver Code Starts
# Initial Template for Python 3


t = int(input())
for _ in range(0, t):
    n = int(input())
    #    l=list(map(int,input().split()))
    #    n=l[0]
    #    m=l[1]
    a = list(map(int, input().split()))
    # k = int(input())
    #  b = list(map(int, input().split()))
    ob = Solution()
    ans = ob.minNumber(a, n)
    print(ans)

# } Driver Code Ends
